OrbitGl/App.cpp

```diff
@@ -466,12 +467,49 @@ void OrbitApp::SetClipboard(const std::string& text) {
 }
 
 ErrorMessageOr<void> OrbitApp::OnSavePreset(const std::string& filename) {
-  OUTCOME_TRY(Capture::SavePreset(filename));
+  OUTCOME_TRY(SavePreset(filename));
   ListPresets();
   Refresh(DataViewType::PRESETS);
   return outcome::success();
 }
 
+ErrorMessageOr<void> OrbitApp::SavePreset(const std::string& filename) {
+  PresetInfo preset;
+  const int32_t pid = m_ProcessesDataView->GetSelectedProcessId();
+  const std::shared_ptr<Process>& process = FindProcessByPid(pid);
+  preset.set_process_full_path(
+      data_manager_->GetProcessByPid(pid)->full_path());
+
+  for (uint64_t function_address : data_manager_->selected_functions()) {
+    FunctionInfo* func = process->GetFunctionFromAddress(function_address);
+    // No need to store the manually instrumented functions
+    if (!FunctionUtils::IsOrbitFunc(*func)) {
+      uint64_t hash = FunctionUtils::GetHash(*func);
+      (*preset.mutable_path_to_module())[func->loaded_module_path()]
+          .add_function_hashes(hash);
+    }
+  }
+
+  std::string filename_with_ext = filename;
+  if (!absl::EndsWith(filename, ".opr")) {
+    filename_with_ext += ".opr";
+  }
+
+  std::ofstream file(filename_with_ext, std::ios::binary);
+  if (file.fail()) {
+    ERROR("Saving preset in \"%s\": %s", filename_with_ext, "file.fail()");
+    return ErrorMessage("Error opening the file for writing");
+  }
+
+  {
+    SCOPE_TIMER_LOG(
```

Comment:
Maybe we want to log this, but probably just normally, I don't think we need to time it.

Reply:
Done.